sequencer/src/block/queryable.rs

```diff
@@ -64,6 +64,51 @@ impl QueryablePayload for Payload<TxTableEntryWord> {
         TxIterator::new(ns_table, self)
     }
 
+    fn transaction(
+        &self,
+        meta: &Self::Metadata,
+        index: &Self::TransactionIndex,
+    ) -> Option<Self::Transaction> {
+        let index_usize = index.tx_idx; // TODO fix in https://github.com/EspressoSystems/espresso-sequencer/issues/1010
+        if index_usize >= self.len(meta) {
+            return None; // error: index out of bounds
+        }
+
+        // start
+        let tx_table_range_start = if index_usize == 0 {
+            None
+        } else {
+            let range_proof_start = index_usize.checked_mul(TxTableEntry::byte_len())?;
+            Some(TxTableEntry::from_bytes(self.raw_payload.get(
```

Comment:
Similarly whenever we read from the `raw_payload` don't we need to offset by the position of the namespace in the payload, from the namespace table?